flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/polling/PollingRecordPublisher.java

```diff
@@ -168,8 +177,8 @@ private GetRecordsResult getRecords(String shardItr, int maxNumberOfRecords)
 
                 // sleep for the fetch interval before the next getRecords attempt with the
                 // refreshed iterator
-                if (expiredIteratorBackoffMillis != 0) {
-                    Thread.sleep(expiredIteratorBackoffMillis);
+                if (fetchIntervalMillis != 0) {
+                    Thread.sleep(fetchIntervalMillis);
```

Comment:
Why not put `Thread.sleep(fetchIntervalMillis);` after `kinesisProxy.getRecords`?
If we put in  catch . The kinesis read still too fast.  `fetchIntervalMillis` seem have nothing about exception and just sleep.

Reply:
It is called after `GetRecords` in the `adjustRunLoopFrequency` at line 216. It does not simply sleep for `fetchIntervalMillis`, it tries to make the total loop time `fetchIntervalMillis`. For instance, if `GetRecords` takes 200ms and the `fetchIntervalMillis` is set to 1000ms, it would sleep for 800ms.

Reply:
I mean why not 

```
try {
                getRecordsResult = kinesisProxy.getRecords(shardItr, maxNumberOfRecords);
                if (fetchIntervalMillis != 0) {
                    Thread.sleep(fetchIntervalMillis);
```

instead to putting the sleep in catch?

Reply:
If we did that, then we would not sleep in the error case. But for the happy path it is equivalent, since the `Thread.sleep(..)` is called by the parent method.

Reply:
I got it. Thanks